### PR TITLE
Zeus Suppression Module

### PR DESCRIPTION
**Pull request by bux**
**When merged this pull request will:**
- Add a module that will allow a zeus to direct suppression fire to certain location / unit

@PabstMirror did all the real work

_click to enlarge_
[](https://i.gyazo.com/b77353e2b44e290f72871f783e444c9a.gif)

### CODE REVIEW

**Comment by bux** [comment]
I guess the only thing that's left, is to disable the line of sight check for indirect fire units (mortars / artillery

File: addons/zeus/functions/fnc_moduleSuppressiveFireLocal.sqf [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        _burstsLeft = _burstsLeft - 1;
+        _this set [1, _burstsLeft];
+        if (_artilleryMag == "") then {
+            _unit doSuppressiveFire _targetASL;
+            TRACE_2("doSuppressiveFire",_unit,_targetASL);
+        _this set [2, _nextRun + 4];
+        } else {
+            (vehicle _unit) doArtilleryFire [ASLtoAGL _targetASL, _artilleryMag, 1]; // note, using value greater than 1 does not always work
+            TRACE_3("doArtilleryFire",_unit,_targetASL,_artilleryMag);
+        _this set [2, _nextRun + 7];
```
Comment (Glowbal) [inline addons/zeus/functions/fnc_moduleSuppressiveFireLocal.sqf]: indentation is off for line 37 and line 33.

File: addons/zeus/functions/fnc_moduleSuppressiveFireLocal.sqf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ * 2: Artiller Magazine <STRING>
+ *
+ * Return Value:
+ * None
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+#define
```
Comment (AACO) [inline addons/zeus/functions/fnc_moduleSuppressiveFireLocal.sqf]: empty `#define` ?

**Comment by SilentSpike** [comment]
Just saw this for the first time, really elegant work with the module target selection @PabstMirror  :+1: Extra happy you managed to generalize it so it can be used in other modules too

**Pull request merged**